### ISSUE TITLE
Throw exception when gradle version used is newer then max supported version

### ISSUE THREAD

**Issue by emanuelb** [bitcoin-wallet/bitcoin-wallet#894]
Add code as below to `settings.gradle` file to abort when gradle version used is newer then the latest gradle version that can build the app (latest version was built successfully with gradle 5.6.4)

```gradle
def maxGradleVersion = GradleVersion.version("5.6.4")

if (GradleVersion.current() > maxGradleVersion)
    throw new GradleScriptException("build requires Gradle older then ${maxGradleVersion} and newer then ${minGradleVersion}", null)
```

The file has a check for using too old version at:
https://github.com/bitcoin-wallet/bitcoin-wallet/blob/f2b6d7cc6ffe690b0593153f5169e4fdbd6fb93e/settings.gradle#L4-L8

Suggestion is to add check for using too new version as well, doing so will return a error message that explain the need of using older gradle version as otherwise the error returned from newer gradle versions is not clear, examples:

I tried to compile this app in the past with gradle 6 in #696  that result in androidx error
Another user reported different build error in method compile() when using gradle 7 in #735
And I recently built with gradle 7.6 that return different error:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/appuser/app/schildbachBitcoin/bitcoin-wallet/wallet/build.gradle' line: 122

* What went wrong:
A problem occurred evaluating project ':wallet'.
> Value is null
```

**Comment by schildbach** [comment]
I general, I think adding a check for the maximum version is a good idea. However, for me the highest working version seems to be 6.x, with 7.x failing the `:wallet:svgToPngMipmap` task.

I just successfully checked these versions: 5.2.1, 5.3, 5.6.4, 6.1.1, 6.7.1, 6.9.2
I have yet to check the versions mentioned in #696: 6.7.1, 6.8.3

**Comment by schildbach** [comment]
For the records: I just also tested 6.8.3 and 6.9.3 – work fine!

**Comment by schildbach** [comment]
See #897

**Comment by schildbach** [comment]
Fixed via #897.
